packages/angular/src/plugins/plugin.ts

```diff
@@ -0,0 +1,733 @@
+  architect?: Record<string, AngularTargetConfiguration>;
+  targets?: Record<string, AngularTargetConfiguration>;
+};
+type AngularJson = { projects?: Record<string, AngularProjectConfiguration> };
+
+const knownExecutors = {
+  appShell: new Set(['@angular-devkit/build-angular:app-shell']),
+  build: new Set([
+    '@angular-devkit/build-angular:application',
+    '@angular/build:application',
+    '@angular-devkit/build-angular:browser-esbuild',
+    '@angular-devkit/build-angular:browser',
+    '@angular-devkit/build-angular:ng-packagr',
+  ]),
+  devServer: new Set(['@angular-devkit/build-angular:dev-server']),
+  extractI18n: new Set(['@angular-devkit/build-angular:extract-i18n']),
+  prerender: new Set(['@angular-devkit/build-angular:prerender']),
+  server: new Set(['@angular-devkit/build-angular:server']),
+  serveSsr: new Set(['@angular-devkit/build-angular:ssr-dev-server']),
+  test: new Set(['@angular-devkit/build-angular:karma']),
+};
+
+const pmc = getPackageManagerCommand();
+
+function readProjectsCache(cachePath: string): Record<string, AngularProjects> {
+  return existsSync(cachePath) ? readJsonFile(cachePath) : {};
+}
+
+function writeProjectsToCache(
+  cachePath: string,
+  results: Record<string, AngularProjects>
+) {
+  writeJsonFile(cachePath, results);
+}
+
+export const createNodesV2: CreateNodesV2<AngularPluginOptions> = [
+  '**/angular.json',
+  async (configFiles, options, context) => {
+    const optionsHash = hashObject(options);
+    const cachePath = join(
+      workspaceDataDirectory,
+      `angular-${optionsHash}.hash`
+    );
+    const projectsCache = readProjectsCache(cachePath);
+    try {
+      return await createNodesFromFiles(
+        (configFile, options, context) =>
+          createNodesInternal(configFile, options, context, projectsCache),
+        configFiles,
+        options,
+        context
+      );
+    } finally {
+      writeProjectsToCache(cachePath, projectsCache);
+    }
+  },
+];
+
+async function createNodesInternal(
+  configFilePath: string,
+  options: AngularPluginOptions,
+  context: CreateNodesContextV2,
+  projectsCache: Record<string, AngularProjects>
+): Promise<CreateNodesResult> {
+  const angularWorkspaceRoot = dirname(configFilePath);
+
+  // Do not create a project if package.json isn't there
+  const siblingFiles = readdirSync(
+    join(context.workspaceRoot, angularWorkspaceRoot)
+  );
+  if (!siblingFiles.includes('package.json')) {
+    return {};
+  }
+
+  const hash = await calculateHashForCreateNodes(
+    angularWorkspaceRoot,
+    options,
+    context,
+    [getLockFileName(detectPackageManager(context.workspaceRoot))]
+  );
+
+  projectsCache[hash] ??= await buildAngularProjects(
+    configFilePath,
+    angularWorkspaceRoot,
+    options,
+    context
+  );
+
+  return { projects: projectsCache[hash] };
+}
+
+async function buildAngularProjects(
+  configFilePath: string,
+  angularWorkspaceRoot: string,
+  options: AngularPluginOptions,
```

Comment:
I don't see this being used

Reply:
Yeah, it's related to having everything wired up for/if options are needed at some point, but we could remove it for now.